### ISSUE TITLE
dogstatsd - error - unparseable metric packet

### ISSUE THREAD

**Issue by mulloymorrow** [DataDog/dd-agent#2523]
**Stacktrace:**
```
2016-05-20 02:10:48 UTC | ERROR | dd.dogstatsd | dogstatsd(dogstatsd.py:361) | Error receiving datagram
Traceback (most recent call last):
  File "/opt/datadog-agent/agent/dogstatsd.py", line 349, in start
    aggregator_submit(message)
  File "/opt/datadog-agent/agent/aggregator.py", line 604, in submit_packets
    parsed_packets = self.parse_metric_packet(packet)
  File "/opt/datadog-agent/agent/aggregator.py", line 440, in parse_metric_packet
    raise Exception('Unparseable metric packet: %s' % packet)
Exception: Unparseable metric packet: X
```

**Setup:**
https://github.com/DataDog/docker-dd-agent 

- `docker run -d --name dd-agent -v /var/run/docker.sock:/var/run/docker.sock -v /proc/:/host/proc/:ro -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro -e API_KEY={your_api_key_here} datadog/docker-dd-agent`
- `docker exec -it <CONTAINER_ID> bash`
- `tail -f /var/log/datadog/dogstatsd.log`

Then send the agent some datagrams with the following command:
- `echo -n "DATGRAM" | nc -vzu localhost 8125`

**Example datagrams** from documentation:
http://docs.datadoghq.com/guides/dogstatsd/ 
```
# Increment the page.views counter.
page.views:1|c

# Record the fuel tank is half-empty
fuel.level:0.5|g

# Sample a the song length histogram half of the time.
song.length:240|h|@0.5

# Track a unique visitor to the site.
users.uniques:1234|s

# Increment the users online counter tagged by country of origin.
users.online:1|c|#country:china

# An example putting it all together.
users.online:1|c|@0.5|#country:china
```

**Comment by remh** [comment]
Thanks for the feedback @mulloymorrow 
I think it's caused by the use of the `z` flag in netcat.
According to the docs:
``` -z      Specifies that nc should just scan for listening daemons, without sending any data to them.  It is an error to use this option in conjunction with the -l option.```

So you should remove it.

I had no issue when running commands such as:

```
echo -n "users.online:1|c|@0.5|#country:china" | nc -vu -w1 localhost 8125
```

If you send `DATGRAM` as a datagram though you'll get an `Unparseable metric packet` exception as it won't be in the expected format.

I'm closing this issue for now.

**Comment by mulloymorrow** [comment]
Excellent! Thank you

On Fri, May 20, 2016, 8:23 AM Remi Hakim <notifications@github.com> wrote:

> Closed #2523 <https://github.com/DataDog/dd-agent/issues/2523>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/DataDog/dd-agent/issues/2523#event-667286391>
>
-- 

Best,
Mulloy Morrow
858.598.3059
